packages/flutter/lib/src/widgets/animated_grid.dart

```diff
@@ -0,0 +1,714 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+
+import 'basic.dart';
+import 'framework.dart';
+import 'scroll_controller.dart';
+import 'scroll_physics.dart';
+import 'scroll_view.dart';
+import 'sliver.dart';
+import 'ticker_provider.dart';
+
+/// Signature for the builder callback used by [AnimatedGrid].
+typedef AnimatedGridItemBuilder = Widget Function(BuildContext context, int index, Animation<double> animation);
+
+/// Signature for the builder callback used by [AnimatedGridState.removeItem].
+typedef AnimatedGridRemovedItemBuilder = Widget Function(BuildContext context, Animation<double> animation);
```

Comment:
Since the signatures are the same, do you think there is any benefit of creating one AnimatedItemBuilder and one AnimatedRemovedItemBuilder that both list and grid use?
We would probably have to add the new ones to AnimatedList, and deprecate the old ones... so if it feels like unnecessary churn we can have separate ones.

Reply:
I can explore this in the follow up refactor. I'll file an issue to track.

Reply:
Renamed to `AnimatedItemBuilder`, and we'll refactor `AnimatedList` later to take that instead.